pulp_ansible/app/galaxy/v3/views.py

```diff
@@ -434,24 +439,55 @@ def get_queryset(self):
         """
         distro_content = self._distro_content
 
-        return CollectionVersion.objects.select_related("content_ptr__contentartifact").filter(
-            pk__in=distro_content
+        return CollectionVersion.objects.filter(pk__in=distro_content).values(
+            "pk",
+            "name",
+            "namespace",
+            "version",
+            "requires_ansible",
+            "authors",
+            "contents",
+            "dependencies",
+            "description",
+            "documentation",
+            "homepage",
+            "issues",
+            "license",
+            "repository",
+            "collection_id",
+            "collection__pulp_created",
+            "collection__pulp_last_updated",
+            "contentartifact__artifact__sha256",
+            "contentartifact__artifact__size",
+        )
+
+    @staticmethod
+    def construct_full_collection_version(obj):
+        """Constructs a new CollectionVersion object."""
+        collection_attrs = {
+            "pk": obj["collection_id"],
+            "pulp_created": obj.pop("collection__pulp_created"),
+            "pulp_last_updated": obj.pop("collection__pulp_last_updated"),
+            "name": obj["name"],
+            "namespace": obj["namespace"],
+        }
+        collection = Collection(**collection_attrs)
```

Comment:
I don't know much about django ORM, but this method seems to be rebuilding some django method

Reply:
I typically use cprofile to get python code timings and visualize them. (lmk if you want any help).

Also getting the SQL EXPLAIN statements for the SQL produced is then more easily analyzed to show 'oh look these queries are better/faster'.